processing/data_collection/gazette/spiders/df_brasilia.py

```diff
@@ -0,0 +1,85 @@
+import re
+import json
+import itertools
+import dateparser
+
+from datetime import datetime
+from scrapy import Request
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class DfBrasiliaSpider(BaseGazetteSpider):
+    TERRITORY_ID = "5300108"
+    GAZETTE_URL = "http://dodf.df.gov.br/listar"
+
+    MONTHS = {
+        "12": "12_Dezembro",
+        "11": "11_Novembro",
+        "10": "10_Outubro",
+        "09": "09_Setembro",
+        "08": "08_Agosto",
+        "07": "07_Julho",
+        "06": "06_Junho",
+        "05": "05_Maio",
+        "04": "04_Abril",
+        "03": "03_Março",
+        "02": "02_Fevereiro",
+        "01": "01_Janeiro",
+    }
+    YEARS = ["2018", "2017", "2016", "2015"]
+    MONTHS_YEARS = itertools.product(MONTHS.values(), YEARS)
+
+    DATE_REGEX = r"DODF [0-9]+ ([0-9]{2}-[0-9]{2}-[0-9]{4})(.*)$"
+    EXTRA_EDITION_TEXT = "EDICAO EXTR"
+    PDF_URL = "http://dodf.df.gov.br/index/visualizar-arquivo/?pasta={}&arquivo={}"
+
+    allowed_domains = ["dodf.df.gov.br"]
+    name = "df_brasilia"
+
+    def start_requests(self):
+        for month, year in self.MONTHS_YEARS:
```

Comment:
This should be enough to get the list of years available (from http://dodf.df.gov.br/listar):
`response.css('#local-arquivos .arquivo::attr(data-file)').extract()`.

For each year, you can yield a request to `http://dodf.df.gov.br/listar?dir={YEAR}`, and using the same selector as above, you can get the list of the months available in that year, so you can yield the request in the following line and use the `parse_month`as it is now.